lib/dependency_parser/ruby/parse.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+require 'bundler'
+require 'json'
+
+module DependencyParser
+  module Ruby
+    class Parse
+      def initialize(content)
+        @content = content
+      end
+
+      def call
+        deps = Bundler::LockfileParser.new(content)
+        @direct = deps.specs.map do |spec|
+          fetch_spec(name: spec.name, version: spec.version)
+        end.compact
+      end
+
+      def direct
+        { repos: @direct, language: "ruby" }
+      end
+
+      private
+
+      attr_reader :content
+
+      def self.valid_gemfile?(content)
+        Bundler::LockfileParser.new(content).specs.any?
+      rescue
+        false
+      end
+
+      def fetch_spec(name:, version: nil)
+        full_spec = fetcher.fetch_spec([name, version])
+        { name: full_spec.name, url: extract_url(full_spec), description: full_spec.description }
+      rescue
```

Comment:
Was there a class of error that triggered this specific rescue? I'm just curious. The other rescue looks like a "passed in the wrong file" kind of thing. I feel like we should capture this error too, but I'm not entirely sure what the failure modes that would trigger the rescue here are.

Reply:
I haven't found any. I can remove it, but it seems like it's better to log the failed spec instead of failing the entire file, we can at least get some log from it.